packages/vitest/src/runtime/mocker.ts

```diff
@@ -32,8 +32,7 @@ type Key = string | symbol
 
 function isSpecialProp(prop: Key, parentType: string) {
   return parentType.includes('Function')
-      && typeof prop === 'string'
-      && ['arguments', 'callee', 'caller', 'length', 'name'].includes(prop)
+      && ['arguments', 'callee', 'caller', 'length', 'name', Symbol.hasInstance].includes(prop)
```

Comment:
I don't like this solution. There are valid cases where we _do_ want to mock `Symbol.hasInstance`, and, as you've noticed, this doesn't actually fix the root cause.